src/Common/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -37,366 +31,296 @@
+export surface_conditions, 
+    compute_R_z0,
+    compute_exchange_coefficients,
+    compute_f_h,
+    compute_f_m,
+    compute_Psi,
+    compute_psi,
+    Momentum,
+    Heat
 
-""" Computes ψ_h for stable case. See Eq. 13 """
-ψ_h_stable(ζ, ζ_0, β_h) = -β_h * (ζ - ζ_0)
+"""
+    SurfaceFluxConditions{FT}
 
-""" Computes ψ_m for unstable case. See Eq. 14 """
-function ψ_m_unstable(ζ, ζ_0, γ_m)
-    x(ζ) = sqrt(sqrt(1 - γ_m * ζ))
-    temp(ζ, ζ_0) = log1p((ζ - ζ_0) / (1 + ζ_0))  # log((1 + ζ)/(1 + ζ_0))
-    ψ_m = (
-        2 * temp(x(ζ), x(ζ_0)) + temp(x(ζ)^2, x(ζ_0)^2) - 2 * atan(x(ζ)) +
-        2 * atan(x(ζ_0))
-    )
-    return ψ_m
-end
+Surface flux conditions, returned from `surface_conditions`.
 
-""" Computes ψ_h for unstable case. See Eq. 15 """
-function ψ_h_unstable(ζ, ζ_0, γ_h)
-    y(ζ) = sqrt(1 - γ_h * ζ)
-    ψ_h = 2 * log1p((y(ζ) - y(ζ_0)) / (1 + y(ζ_0))) # log((1 + y(ζ))/(1 + y(ζ_0)))
-    return ψ_h
-end
+# Fields
 
+$(DocStringExtensions.FIELDS)
 """
-    monin_obukhov_len(param_set, u, flux)
+struct SurfaceFluxConditions{FT}
+    L_MO::FT
+    pottemp_flux_star::FT
+    flux::Vector{FT}
+    x_star::Vector{FT}
+    K_exchange::Vector{FT}
+end
 
-Computes the Monin-Obukhov length (Eq. 3)
-"""
-function monin_obukhov_len(param_set::AbstractParameterSet, u, flux)
-    FT = typeof(u)
-    _von_karman_const::FT = von_karman_const(param_set)
-    return -u^3 / (flux * _von_karman_const)
+function Base.show(io::IO, sfc::SurfaceFluxConditions)
+    println(io, "----------------------- SurfaceFluxConditions")
+    println(io, "L_MO              = ", sfc.L_MO)
+    println(io, "pottemp_flux_star = ", sfc.pottemp_flux_star)
+    println(io, "flux              = ", sfc.flux)
+    println(io, "x_star            = ", sfc.x_star)
+    println(io, "K_exchange        = ", sfc.K_exchange)
+    println(io, "-----------------------")
 end
 
 """
-    compute_friction_velocity(param_set, flux, z_0, z_1, β_m, γ_m, tol_abs, iter_max)
-
-Computes roots of friction velocity equation (Eq. 10)
-
-`u_ave = u_* ( ln(z/z_0) - ψ_m(z/L, z_0/L) ) /κ`
+    surface_conditions
+Surface conditions given
+ - `x_initial` initial guess for solution (`L_MO, u_star, θ_star, ϕ_star, ...`)
+ - `x_ave` volume-averaged value for variable `x`
+ - `x_s` surface value for variable `x`
+ - `z_0` roughness length for variable `x`
+ - `F_exchange` flux at the top for variable `x`
+ - `dimensionless_number` dimensionless turbulent transport coefficient:
+      - Momentum: 1
+      - Heat: Turbulent Prantl number at neutral stratification
+      - Mass: Turbulent Schmidt number
+      - ...
+ - `θ_bar` basic potential temperature
+ - `Δz` layer thickness (not spatial discretization)
+ - `z` coordinate axis
+ - `a` free model parameter with prescribed value of 4.7
+ - `pottemp_flux_given` potential temperature flux (optional)
+
+If `pottemp_flux` is not given, then it is computed by iteration
+of equations 3, 17, and 18 in Nishizawa2018.
```

Comment:
But we are not using the finite-volume similarity functions in Nishizawa, but some other (Businger?) functions for nodal values, correct?